### PR TITLE
[WIP] Move query function from db_test to storage.go

### PR DESCRIPTION
**Pull request by aSquare14**
Signed-off-by: aSquare14 <atibhi.a@gmail.com>
Moved query function from db_test.go to a new file in testutils because it needs to be reused in backfilling PR. #8084

### CODE REVIEW

File: util/testutil/storage.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+// limitations under the License.
+
+package testutil
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb/tsdbutil"
```
Comment (brian-brazil) [inline util/testutil/storage.go]: I don't think that testutil should be depending on storage, as testutil is used by things like discovery that are completely unrelated to the tsdb.

Comment (beorn7) [reply util/testutil/storage.go]: What do you suggest instead?

Comment (brian-brazil) [reply util/testutil/storage.go]: It'd have to go somewhere else to keep the code well factored, what's the goal of this change? We have a teststorage already, or maybe in tsdbutil considering it's using a struct.

Comment (beorn7) [reply util/testutil/storage.go]: > what's the goal of this change?

To make the `query` function used in tests usable outside of the tsdb package. That's implied by the description above:

"…because it needs to be reused in backfilling PR. #8084 "

Comment (beorn7) [reply util/testutil/storage.go]: `teststorage` might be a good location for this. It would, however, have nothing to do with the actual `TestStorage` struct in there, but it would still have to do with testing the storage. I would be fine with that location. (And then perhaps move `queryChunks`, too, for consistency.)

`tsdbutil` sounds good, too. I was reluctant at first because the functions in question here are only used in tests. But that's true for some types in `tsdbutil`, too.

@aSquare14 you see, there are plenty of possibilities. However, you have to be aware of import cycles. The PR as is is creating an import cycle:

```
# github.com/prometheus/prometheus/tsdb/fileutil
package github.com/prometheus/prometheus/tsdb/fileutil (test)
        imports github.com/prometheus/prometheus/util/testutil
        imports github.com/prometheus/prometheus/storage
        imports github.com/prometheus/prometheus/tsdb/chunks
        imports github.com/prometheus/prometheus/tsdb/fileutil: import cycle not allowed in test
```

This particular one should be avoided by moving into `tsdbutil` or `teststorage`, but others might arise…

**Comment by aSquare14** [comment]
I am working on the build failures and will fix whatever has been addressed. This PR is still a WIP, shall re-request for review after I fix.

**Comment by beorn7** [comment]
Just for context: After @aSquare14 refined the tests, she needed a different helper function, so the original reason for this PR disappeared. If we ever need something like this for testing outside of TSDB, we know now where to put it.
